characters/character_routes.py

```diff
@@ -162,4 +172,113 @@ def make_character_feat_mapping():
+    #check if the integer they sent was a character that exists
+    character = characters.get_character(sani_character_pk_id)
+    if character is None:
+        flash("That Character doesn't seem to exist.")
+        log.warn("%s attempted to give a skill to a character that doesn't exist. Character id %s." % (session['username'], pk_id))
+        return redirect("/characters/mine")
+    #check if the integer they sent was a character that they own
+    user_id = security.get_user_pkid(session)
+    if user_id != character['owner']:
+        flash("You don't own that character!")
+        log.warn("%s attempted to give a skill to a character that they don't own! Character id: %s" % (session['username'], pk_id))
+        return redirect("/")
+    #looks good. make a new skill and send its ID back to 'em.
+    new_skill_pk_id = skills.get_newest_skill(sani_character_pk_id)
+    return "%s" % new_skill_pk_id
+
+""" The pk_id here is for the skill that you wish to delete. Note that is a different 
+    convention from the creation of a skill above. """
+@character_blueprint.route("/skills/delete/<pk_id>", methods=['POST'])
+def delete_skill():
+    #check if the user is logged in
+    if not security.check_auth(session):
+        flash("You can't do that without logging in")
+        return redirect("/")
+    #check if what the user sent was actually an integer
+    try:
+        sani_skill_pk_id = int(pk_id)
+    except:
+        flash("There was an error figuring out which character to delete this skill from.")
+        log.error("There was an error deleting skill from: %s character pk_id." % pk_id)
+        return redirect("/")
+    #check if the integer they sent was a skill that exists
+    skill = skills.get_skill(sani_skill_pk_id)
+    if skill is None:
+        flash("That skill doesn't seem to exist")
+        log.warn("%s attempted to delete a skill that doesn't exist. skill id: %s" % (session['username'], pk_id))
+        return redirect("/characters/mine")
+    character = characters.get_character(skill['fk_owner_id'])
+    if character is None:
+        flash("That Character doesn't seem to exist.")
+        log.warn("%s attempted to delete a skill to a character that doesn't exist. Character id %s." % (session['username'], pk_id))
+        return redirect("/characters/mine")
+    #check if the integer they sent was a character that they own
+    user_id = security.get_user_pkid(session)
+    if user_id != character['owner']:
+        flash("You don't own that character!")
+        log.warn("%s attempted to delete a skill to a character that they don't own! Character id: %s" % (session['username'], pk_id))
+        return redirect("/")
+    #looks good. delete that skill.
+    skills.delete_skill(sani_skill_pk_id)
+    return redirect("/modifycharacter/%s" % character['pk_id'])
+
+
+""" pk_id here is the pk_id of the skill to update. """
+@character_blueprint.route("/skills/modify/<pk_id>", methods=['POST'])
+def update_skill(pk_id):
```

Comment:
Blocker: The log entries in this function reference deletion instead of updating. Looks like copy-pasta chaos.